### ISSUE TITLE
use yaml file for AutomationHAT GPIO pin number assignment

### ISSUE THREAD

**Issue by lspitler** [AstroHuntsman/huntsman-dome#22]
To change the GPIO pin assignments currently one has to hack the `__init__` function of:
https://github.com/AstroHuntsman/huntsman-dome/blob/develop/domehunter/__init__.py

Better to specify a yaml file that contains GPIO pin assignments.

Optional: see this file for how to implement GPIO pin assignments as a separate class:
https://github.com/panoptes/POCS/blob/develop/pocs/dome/astrohaven.py

**Comment by AnthonyHorton** [comment]
@wtgee might encourage us to use the config server from `panoptes-utils` for this: https://github.com/panoptes/panoptes-utils/tree/develop/panoptes/utils/config

**Comment by wtgee** [comment]
Yes, we'll want to figure out if the config server is the best option for
Huntsman. I think it could be as it simplifies a lot, but there's some
question of whether we want to use it in a piecemeal fashion or want one
giant config server.

**Comment by lspitler** [comment]
Interesting idea. Normally for such a simple subsystem, I'd say its overkill, but it is an elegant solution in terms of long-term maintainability. I gather we'd be running a server for the Lens Pis anyways?

@wtgee I guess you must know you're well past the point where it must be tempting to fork off e.g. `config-server` (and a lot of your other utils) as its own repo (`pyamlserver`?) so someone doesn't need to install all of POCs to use this handy util.

**Comment by wtgee** [comment]
On Wed, Aug 14, 2019, 04:39 Lee Spitler <notifications@github.com> wrote:

> Interesting idea. Normally for such a simple subsystem, I'd say its
> overkill, but it is an elegant solution in terms of long-term
> maintainability. I gather we'd be running a server for the Lens Pis anyways?
>
> @wtgee <https://github.com/wtgee> I guess you must know you're well past
> the point where it must be tempting to fork off e.g. config-server (and a
> lot of your other utils) as its own repo (pyamlserver?) so someone
> doesn't need to install all of POCs to use this handy util.
>
It has been split out of POCS for a while now and is a part of
panoptes-utils although I have been tempted to split from that as well.

**Comment by fergusL** [comment]
Should also read in calibrated az_to_ticks from yaml file instead of triggering a calibration run when gotoaz() is called and dome_az is None.
see original comment from @AnthonyHorton 
https://github.com/AstroHuntsman/huntsman-dome/pull/26#discussion_r328411004
